### ISSUE TITLE
Accessing (shell) environment variables in dune files

### ISSUE THREAD

**Issue by prateek-khandelwal** [ocaml/dune#1506]
Hi, 

This might sound elementary, however, I am not able to figure out how one would go about using an shell-environment variable to specify the path for c-header files required in c-stubs compilation workflow. 

Ideally, I would like to specify something like this: `(c_flags (-I<BASE>/foo/bar/include))`, where `BASE` is a variable available in shell's environment. I am sure this can be achieved by using the `Configurator` module, however, I was wondering if this can be done without it.

**Comment by rgrinberg** [comment]
https://github.com/ocaml/dune/pull/1503 will make it possible. E.g:

```
(c_flags (-I%{env:BASE=default}/foo/bar/include))
```

Feel free to give this a try if you'd like.

**Comment by prateek-khandelwal** [comment]
Thanks for the pointer!
I take it that the pull request has not been merged with the master branch yet, is that correct ?

**Comment by ejgallego** [comment]
This should be closed right?

However I couldn't find documentation for `%{env:...}` anywhere, is that a supported variable expansion?

**Comment by rgrinberg** [comment]
It's supported just not everywhere in dune 1.5. Did you include a default value? The format is: `%{env:foo=default}`. In 1.6, we improve the situation so that you can expand environment variables in more places.

**Comment by ejgallego** [comment]
@rgrinberg sorry, I didn't mean it doesn't work, it works fine here for me; what I mean is that this particular expansion seems to be documented nowhere.

**Comment by jonathan-laurent** [comment]
This feature still appears to be undocumented.

**Comment by gasche** [comment]
For the record:
- the feature was initially implemented in #1396 which was included in Dune 1.4.0
- it was expanded slightly and the interaction with `setenv` was fixed in #1503 in Dune 1.5.0
- above @rgrinberg mentions that it should have been improved again in 1.6.0, but I cannot find trace of that (looking at the git log for commits that touch a (test)file matching `*env-var*`, I see no changes later than #1503.)

I was also looking for this feature and I also wasted time looking at the documentation to find it, and then I found this issue, grepped the source and git history to find out, and located those issues. Now I guess I have to submit a documentation PR. Grumble.
